static-site/src/components/Groups/edit-overview-form.jsx

```diff
@@ -0,0 +1,109 @@
+import React, { useEffect, useState } from "react";
+import * as splashStyles from "../splash/styles";
+import { CenteredForm, InputButton, TextArea } from "./styles";
+
+const OVERVIEW_TEMPLATE = `---
+title:
+byline:
+website:
+showDatasets:
+showNarratives:
+---
+
+Add description of your group here.
+`;
+
+const EditOverviewForm = ({ groupName, createErrorMessage, clearErrorMessage }) => {
+  const [ overview, setOverview ] = useState();
+  const [ uploadInProgress, setUploadInProgress ] = useState(false);
+  const [ uploadComplete, setUploadComplete ] = useState(false);
+
+  useEffect(() => {
+    const setCurrentOverview = async () => {
+      const currentOverview = await getOverview();
+      if (!cleanUp) setOverview(currentOverview);
+    }
+
+    let cleanUp = false;
+    setCurrentOverview();
+    return () => cleanUp = true;
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  const getOverview = async () => {
+    clearErrorMessage();
+    try {
+      const response = await fetch(`/groups/${groupName}/settings/overview`);
+      if (response.status === 404) return OVERVIEW_TEMPLATE;
+      if (response.ok) {
+        const currentOverview = await response.text();
+        return currentOverview.trim().length > 0
+          ? currentOverview
+          : OVERVIEW_TEMPLATE;
+      }
+      createErrorMessage(response);
+    } catch (err) {
+      console.error(err.message)
+      createErrorMessage();
+    }
+  }
+
+  const uploadOverview = async (e) => {
+    e.preventDefault();
+    clearErrorMessage();
+    setUploadInProgress(true);
+
+    try {
+      const response = await fetch(`/groups/${groupName}/settings/overview`, {
+        method: "PUT",
+        headers: {
+          "Content-Type": "text/markdown"
+        },
+        body: overview
+      });
+
+      response.ok
+        ? setUploadComplete(true)
+        : createErrorMessage(response);
```

Comment:
Unlike the logo upload, this bit of code doesn't re-fetch the overview after updating it. That means that if you blank the overview and update it, it stays blank instead of the template from above popping in. It takes a page refresh for the template to pop in.

Reply:
Completed the whole round-trip in 349ef2acd7c94891e20c2652935d9dda45eaf13a